pkg/controller/containerizedworkload/containerizedworkload.go

```diff
@@ -0,0 +1,231 @@
+			}
+			if container.LivenessProbe.TimeoutSeconds != nil {
+				kubernetesContainer.LivenessProbe.TimeoutSeconds = *container.LivenessProbe.TimeoutSeconds
+			}
+			if container.LivenessProbe.PeriodSeconds != nil {
+				kubernetesContainer.LivenessProbe.PeriodSeconds = *container.LivenessProbe.PeriodSeconds
+			}
+			if container.LivenessProbe.SuccessThreshold != nil {
+				kubernetesContainer.LivenessProbe.SuccessThreshold = *container.LivenessProbe.SuccessThreshold
+			}
+			if container.LivenessProbe.FailureThreshold != nil {
+				kubernetesContainer.LivenessProbe.FailureThreshold = *container.LivenessProbe.FailureThreshold
+			}
+
+			// NOTE(hasheddan): Kubernetes specifies that only one type of
+			// handler should be provided. OAM does not impose that same
+			// restriction. We optimistically check all and set whatever is
+			// provided.
+			if container.LivenessProbe.HTTPGet != nil {
+				kubernetesContainer.LivenessProbe.Handler.HTTPGet = &corev1.HTTPGetAction{
+					Path: container.LivenessProbe.HTTPGet.Path,
+					Port: intstr.IntOrString{IntVal: container.LivenessProbe.HTTPGet.Port},
+				}
+
+				for _, h := range container.LivenessProbe.HTTPGet.HTTPHeaders {
+					kubernetesContainer.LivenessProbe.Handler.HTTPGet.HTTPHeaders = append(kubernetesContainer.LivenessProbe.Handler.HTTPGet.HTTPHeaders, corev1.HTTPHeader{
+						Name:  h.Name,
+						Value: h.Value,
+					})
+				}
+			}
+			if container.LivenessProbe.Exec != nil {
+				kubernetesContainer.LivenessProbe.Exec = &corev1.ExecAction{
+					Command: container.LivenessProbe.Exec.Command,
+				}
+			}
+			if container.LivenessProbe.TCPSocket != nil {
+				kubernetesContainer.LivenessProbe.TCPSocket = &corev1.TCPSocketAction{
+					Port: intstr.IntOrString{IntVal: container.LivenessProbe.TCPSocket.Port},
+				}
+			}
+		}
+
+		if container.ReadinessProbe != nil {
+			kubernetesContainer.ReadinessProbe = &corev1.Probe{}
+			if container.ReadinessProbe.InitialDelaySeconds != nil {
+				kubernetesContainer.ReadinessProbe.InitialDelaySeconds = *container.ReadinessProbe.InitialDelaySeconds
+			}
+			if container.ReadinessProbe.TimeoutSeconds != nil {
+				kubernetesContainer.ReadinessProbe.TimeoutSeconds = *container.ReadinessProbe.TimeoutSeconds
+			}
+			if container.ReadinessProbe.PeriodSeconds != nil {
+				kubernetesContainer.ReadinessProbe.PeriodSeconds = *container.ReadinessProbe.PeriodSeconds
+			}
+			if container.ReadinessProbe.SuccessThreshold != nil {
+				kubernetesContainer.ReadinessProbe.SuccessThreshold = *container.ReadinessProbe.SuccessThreshold
+			}
+			if container.ReadinessProbe.FailureThreshold != nil {
+				kubernetesContainer.ReadinessProbe.FailureThreshold = *container.ReadinessProbe.FailureThreshold
+			}
+
+			// NOTE(hasheddan): Kubernetes specifies that only one type of
+			// handler should be provided. OAM does not impose that same
+			// restriction. We optimistically check all and set whatever is
+			// provided.
```

Comment:
I was thinking when writing out this part of the spec that it might be an improvement to include a type distinctor; i.e. so that we'd only pay attention to `probe.HTTPGet` if `probe.Type == HTTPGet`.